Library/OcBootManagementLib/VariableManagement.c

```diff
@@ -89,8 +90,11 @@ IsDeletableVariable (
     //
     // Only erase boot and driver entries for BDS
     // I.e. BootOrder, Boot####, DriverOrder, Driver####
+    // Preserve Boot#### entries except BootNext when PreserveBoot is TRUE.
     //
-    if (  (  (StrnCmp (Name, L"Boot", StrLen (L"Boot")) == 0)
+    if (  (StrCmp (Name, L"BootNext") == 0)
+       || (  !PreserveBoot
+          && (StrnCmp (Name, L"Boot", StrLen (L"Boot")) == 0)
```

Comment:
Probably for another patchiest, but we should replace all such `StrLen` calls with `L_STR_LEN`

Reply:
Raised separate issue. Can do after this.